php/async/Exchange.php

```diff
@@ -79,8 +79,16 @@ public function set_request_browser($connector_options = array()) {
 
     public function fetch($url, $method = 'GET', $headers = null, $body = null) {
         // wrap this in as a promise so it executes asynchronously
-        return React\Async\async(function () use ($url, $method, $headers, $body) {
+        $fetch_result = $this->fetch_result;
+        return React\Async\async(function () use ($url, $method, $headers, $body, $fetch_result) {
 
+            if ($fetch_result !== null) {
+                // used in tests
+                // it's not the prettiest solution but
+                // couldn't find a better way to do it
+                // feel free to suggest a better alternative
+                return $fetch_result;
```

Comment:
@ttodua I tried a lot of things (dynamic overrides, dynamic class extends, etc) but couldn't find a better alternative in PHP to mock the https response. do you have a better suggestion?

Reply:
i'll check some things and lyk